redbot/cogs/general/general.py

```diff
@@ -224,49 +223,45 @@ def __init__(self):
 
     @commands.command()
-    async def urban(self, ctx, *, search_terms: str, definition_number: int = 1):
-        """Urban Dictionary search
+    async def urban(self, ctx, *, word):
+        """Searches for urban dictionary entries using the unofficial urbandictionary api"""
 
-        Definition number must be between 1 and 10"""
+        data = None
 
-        def encode(s):
-            return quote_plus(s, encoding="utf-8", errors="replace")
+        try:
+            url = "https://api.urbandictionary.com/v0/define?term=" + str(word).lower()
 
-        # definition_number is just there to show up in the help
-        # all this mess is to avoid forcing double quotes on the user
+            headers = {"content-type": "application/json"}
 
-        search_terms = search_terms.split(" ")
-        try:
-            if len(search_terms) > 1:
-                pos = int(search_terms[-1]) - 1
-                search_terms = search_terms[:-1]
-            else:
-                pos = 0
-            if pos not in range(0, 11):  # API only provides the
-                pos = 0  # top 10 definitions
-        except ValueError:
-            pos = 0
-
-        search_terms = {"term": "+".join([s for s in search_terms])}
-        url = "http://api.urbandictionary.com/v0/define"
-        try:
             async with aiohttp.ClientSession() as session:
-                async with session.get(url, params=search_terms) as r:
-                    result = await r.json()
-            item_list = result["list"]
-            if item_list:
-                definition = item_list[pos]["definition"]
-                example = item_list[pos]["example"]
-                defs = len(item_list)
-                msg = "**Definition #{} out of {}:\n**{}\n\n**Example:\n**{}".format(
-                    pos + 1, defs, definition, example
-                )
-                msg = pagify(msg, ["\n"])
-                for page in msg:
-                    await ctx.send(page)
-            else:
-                await ctx.send(_("Your search terms gave no results."))
-        except IndexError:
-            await ctx.send(_("There is no definition #{}").format(pos + 1))
+                async with session.get(url, headers=headers) as response:
+                    data = await response.json()
+
         except:
-            await ctx.send(_("Error."))
+            await ctx.send("No Urban dictionary were found or there was an error in the process")
+
+        if data.get("error") != 404 and data.get("result_type") == "exact":
+
+            # a list of embeds
+            embeds = []
+            for ud in data["list"]:
+                embed = discord.Embed()
+                embed.title = ud["word"].capitalize() + " by " + ud["author"]
+                embed.url = ud["permalink"]
+                embed.description = (
+                    ud["definition"] + "\n \n **Example : **" + ud.get("example", "N/A")
+                )
```

Comment:
Please use `"".format()` where instead of concatenating strings, and also add back in the i18n call (which is surrounding the string with `_()`.